pdr_backend/data_eng/test/test_gql_data_factory.py

```diff
@@ -0,0 +1,287 @@
+    predictions_schema,
+)
+
+from pdr_backend.util.subgraph_predictions import (
+    FilterMode,
+)
+
+# ====================================================================
+# test parquet updating
+pdr_predictions_record = "pdr_predictions"
+
+
+@patch("pdr_backend.data_eng.gql.predictoor.predictions.fetch_filtered_predictions")
+@patch("pdr_backend.data_eng.gql_data_factory.get_all_contract_ids_by_owner")
+def test_update_gql1(
+    mock_get_all_contract_ids_by_owner,
+    mock_fetch_filtered_predictions,
+    tmpdir,
+    sample_daily_predictions,
+):
+    mock_get_all_contract_ids_by_owner.return_value = ["0x123"]
+    _test_update_gql(
+        mock_fetch_filtered_predictions,
+        tmpdir,
+        sample_daily_predictions,
+        "2023-11-02_0:00",
+        "2023-11-04_0:00",
+        n_preds=2,
+    )
+
+
+@patch("pdr_backend.data_eng.gql.predictoor.predictions.fetch_filtered_predictions")
+@patch("pdr_backend.data_eng.gql_data_factory.get_all_contract_ids_by_owner")
+def test_update_gql2(
+    mock_get_all_contract_ids_by_owner,
+    mock_fetch_filtered_predictions,
+    tmpdir,
+    sample_daily_predictions,
+):
+    mock_get_all_contract_ids_by_owner.return_value = ["0x123"]
+    _test_update_gql(
+        mock_fetch_filtered_predictions,
+        tmpdir,
+        sample_daily_predictions,
+        "2023-11-02_0:00",
+        "2023-11-06_0:00",
+        n_preds=4,
+    )
+
+
+@patch("pdr_backend.data_eng.gql.predictoor.predictions.fetch_filtered_predictions")
+@patch("pdr_backend.data_eng.gql_data_factory.get_all_contract_ids_by_owner")
+def test_update_gql3(
+    mock_get_all_contract_ids_by_owner,
+    mock_fetch_filtered_predictions,
+    tmpdir,
+    sample_daily_predictions,
+):
+    mock_get_all_contract_ids_by_owner.return_value = ["0x123"]
+    _test_update_gql(
+        mock_fetch_filtered_predictions,
+        tmpdir,
+        sample_daily_predictions,
+        "2023-11-01_0:00",
+        "2023-11-07_0:00",
+        n_preds=6,
+    )
+
+
+@patch("pdr_backend.data_eng.gql.predictoor.predictions.fetch_filtered_predictions")
+@patch("pdr_backend.data_eng.gql_data_factory.get_all_contract_ids_by_owner")
+def test_update_gql_multiple(
+    mock_get_all_contract_ids_by_owner,
+    mock_fetch_filtered_predictions,
+    tmpdir,
+    sample_daily_predictions,
+):
+    mock_get_all_contract_ids_by_owner.return_value = ["0x123"]
+    _test_update_gql(
+        mock_fetch_filtered_predictions,
+        tmpdir,
+        sample_daily_predictions,
+        "2023-11-02_0:00",
+        "2023-11-04_0:00",
+        n_preds=2,
+    )
+
+    _test_update_gql(
+        mock_fetch_filtered_predictions,
+        tmpdir,
+        sample_daily_predictions,
+        "2023-11-01_0:00",
+        "2023-11-05_0:00",
+        n_preds=3,
+    )
+
+    _test_update_gql(
+        mock_fetch_filtered_predictions,
+        tmpdir,
+        sample_daily_predictions,
+        "2023-11-02_0:00",
+        "2023-11-07_0:00",
+        n_preds=5,
+    )
```

Comment:
DYY violation. Try to turn this into 3 lines.

Reply:
ty, done